### DISCUSSION TITLE
[Collapsible] Does it possible to add an offset?

### DISCUSSION THREAD

**Discussion by nicolas-bourles** [radix-ui/primitives#1709] [Help]
### Overview

Hello,

I need to implement a Collapsible component with a part of the content visible in the beginning. By default, the content is hidden with this component. But I need to create an animation from the part of the content visible to the entire content visible. It doen't work...

I tried to implement a default height for the closed state, like this following example below:

```ts
const open = keyframes`
  0% { height: var(--radix-collapsible-content-height-closed) }
  100% { height: var(--radix-collapsible-content-height) }
`

const close = keyframes`
  0% { height: var(--radix-collapsible-content-height) }
  100% { height: var(--radix-collapsible-content-height-closed) }
`
```

Here is the visual representation of that I need to do:

#### State "closed"

![image](https://user-images.githubusercontent.com/22050486/194571968-544d8149-9f9f-4707-8ec9-ebcae9017675.png)

#### State "opened"

![image](https://user-images.githubusercontent.com/22050486/194572336-31675ed6-b3b4-42eb-a937-9040a9d4a5ec.png)

Thanks a lot by advance for your help.

Best Regards.

**Comment by joaom00** [comment]
Hi @nicolas-bourles,

You can add the `Collapsible.Content` only on the part of the content that will be hidden.

```js
<Collapsible.Root>
    <p>Part of the content</p>

    <Collapsible.Content>
        <p>Rest of the content...</p>
    </Collapsible.Content>
</Collapsible.Root>
```

**Reply by andy-hook** [reply]
@nicolas-bourles Additionally, here's [an example](https://codesandbox.io/s/collapsible-text-ll5cwx?file=/src/App.js) showing how this can be animated.
